frontend/reducers/now_playing_reducer.js

```diff
@@ -0,0 +1,23 @@
+import { RECEIVE_CURRENT_ARTIST } from "../actions/artist_actions";
+import { TOGGLE_PLAY } from "../actions/now_playing_actions";
+
+const nowPlayingReducer = (
+	playState = { isPlaying: false, queue: [] },
+	action
+) => {
+	Object.freeze(playState);
```

Comment:
is this the preferred way now? I remember they use to destructure the objects to create the new state

Reply:
Can you say more? I know the rule is to make sure the old state is not modified, so you have to copy it for the new state. Lines 8 & 9 were how we've been taught to do it; maybe because the `.freeze` clearly shows old state is protected. I know that deconstructing would create a copy too.

Reply:
Yeah I think you're right, I think I like this better.